harness/determined/common/api/_session.py

```diff
@@ -170,6 +172,12 @@ def put(
     ) -> requests.Response:
         return self._do_request("PUT", path, params, json, data, headers, timeout, False)
 
+    def with_retry(self: T, max_retries: GeneralizedRetry) -> T:
+        """Generate a new session with a different retry policy."""
+        new_session = copy.copy(self)
```

Comment:
why do we need to copy and create a new session object instead of just returning `self`?

Reply:
I can't think of where we would want to return `self` rather than `None`, but my intent in creating a new object rather than using a setter is so we don't have to worry about the side effect of changing retry logic in cases where a `Session` is shared.

Did you have something else in mind when returning `self`?

Reply:
ah no that makes sense. i'm just generally sus of importing extra dependencies, but what you're saying makes sense